### PR TITLE
Feature/arangosearch pk endianness

### PR DESCRIPTION
**Pull request by gnusi**
Please note that for legal reasons we require you to sign the 
[Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2011/

File: arangod/IResearch/IResearchViewBlock.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ pks_t::iterator readPKs(
   auto end = keys.end();
 
   for (irs::bytes_ref key; begin != end && it.next(); ) {
-    if (values(it.value(), key) && begin->read(key)) {
+    if (values(it.value(), key) && arangodb::iresearch::DocumentPrimaryKey::read(*begin, key)) {
```
Comment (jsteemann) [inline arangod/IResearch/IResearchViewBlock.cpp]: In this loop, `it` will always be advanced for every iteration, but `begin` will only be advanced conditionally, meaning the two can get out of sync.
That may be exactly what was intended here, but I don't know the exact intention.

Comment (gnusi) [reply arangod/IResearch/IResearchViewBlock.cpp]: yes, that's exactly what is intended. The idea is to have a range of valid PKs in preallocated vector without gaps.

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2016/

**Pull request merged**